kamu-adapter-graphql/tests/tests/test_gql_data.rs

```diff
@@ -14,25 +14,92 @@ use datafusion::arrow::array::*;
 use datafusion::arrow::datatypes::{DataType, Field, Schema};
 use datafusion::arrow::record_batch::RecordBatch;
 use kamu::domain::*;
-use kamu::infra;
-use kamu::testing::{MetadataFactory, ParquetWriterHelper};
+use kamu::infra::utils::s3_context::S3Context;
+use kamu::infra::{self};
+use kamu::testing::{MetadataFactory, MinioServer, ParquetWriterHelper};
 use opendatafabric::*;
+use reqwest::Url;
+use tempfile::TempDir;
 
-async fn create_test_dataset(tempdir: &Path) -> dill::Catalog {
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[allow(dead_code)]
+struct S3 {
+    tmp_dir: tempfile::TempDir,
+    minio: MinioServer,
+    url: Url,
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+async fn run_s3_server() -> S3 {
```

Comment:
We do not have any other tests for queries service at the moment. These are the only ones at this moment that activate the problematic behavior

Reply:
Are you saying they belong in GQL crate, or you want to keep them here only because there are a few helper functions here that makes the tests easy to write?

Reply:
Addressed